source/common/network/utility.cc

```diff
@@ -342,5 +346,30 @@ bool Utility::portInRangeList(const Address::Instance& address, const std::list<
   return false;
 }
 
+absl::uint128 Utility::Ip6ntohl(const absl::uint128& address) {
+  // TODO(ccaraman): Support Ip6ntohl for big-endian.
+  static_assert(ABSL_IS_LITTLE_ENDIAN,
+                "Machines using big-endian byte order is not supported for IPv6.");
+  return flipOrder(address);
+}
+
+absl::uint128 Utility::Ip6htonl(const absl::uint128& address) {
+  // TODO(ccaraman): Support Ip6ntohl for big-endian.
+  static_assert(ABSL_IS_LITTLE_ENDIAN,
+                "Machines using big-endian byte order is not supported for IPv6.");
+  return flipOrder(address);
+}
+
+absl::uint128 Utility::flipOrder(const absl::uint128& input) {
+  absl::uint128 result{0};
+  absl::uint128 data = input;
+  for (int i = 0; i < 16; i++) {
+    result <<= 8;
+    result |= (data & 0x000000000000000000000000000000FF);
+    data >>= 8;
```

Comment:
optional : internally I think we roughly return uint128(htonl(input.low), htonl(input.high)) which obviously only works if htonl is flipping bits.   If this is only going to be called for little endian switching might save some operations but I'm also fine leaving as is since if we add big endian support we'd need rename it something awful like flipOrderForLittleEndian to make it clear :-P

Reply:
I'm going to leave it as is.